app/packages/looker/src/index.ts

```diff
@@ -656,16 +700,19 @@ export abstract class Looker<
           disabled: false,
           reloading: false,
         });
-        worker.removeEventListener("message", listener);
+        labelsWorker.removeEventListener("message", listener);
       }
     };
-    worker.addEventListener("message", listener);
-
-    worker.postMessage({
-      method: "processSample",
-      coloring: this.state.options.coloring,
-      sample,
-      uuid: messageUUID,
+    labelsWorker.addEventListener("message", listener);
+
+    // todo: don't process overlays here: https://github.com/voxel51/fiftyone/issues/1344
+    this.imputeOverlayMaskFromDisk(sample).finally(() => {
+      labelsWorker.postMessage({
```

Comment:
You will need to [transfer](https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage) the array buffers that are created if they are created on the main thread.